### PR TITLE
build(base): use OpenJDK UBI images

### LINKED ISSUE #58: Reports pod fails to start on some clusters due to entrypoint permissions
I also got a bit of an issue when creating a report replica as pod seems to in `CrashLoopBackOff` with logs show:

```bash
exec /deployments/run-java.sh: permission denied
```

I am using Cluster Bot with OCP 4.11.1

_Originally posted by @tthvo in https://github.com/cryostatio/cryostat-operator/issues/499#issuecomment-1324156828_

### PR DESCRIPTION
**Pull request by tthvo**
Fixes #58 

Update base image to OpenJDK UBI images: `registry.access.redhat.com/ubi8/openjdk-17:1.14`.

### CODE REVIEW

**Review by ebaron — COMMENTED** [review]
Hi @tthvo, thanks for doing this. I think you'll also want to change the user id from 1001 to 185, which is what's used in the OpenJDK UBI images. We can turn Jolokia off as well, and we may need to specify the JAR name.

See the diff here for a better idea of these suggestions: https://github.com/andrewazores/quarkus-test/commit/d98247b7d7458322b31aee05976b9835d2f8f2c9

**Comment by tthvo** [comment]
Right, thanks @ebaron, I added in those configurations now. Just a question about `Dockerfile.legacy-jar`? What is this docker file for and do we need to change it too?

**Comment by ebaron** [comment]
This seems to copy the jar artifact to quarkus-run.jar, no? https://github.com/andrewazores/quarkus-test/blob/ebe65aa9ab0e3a2453ea2e36bc233fbf04cbaba4/src/main/docker/Dockerfile.legacy-jar#L84

The base image is indeed out of date. It should use JDK 17 UBI as well. We don't use the legacy-jar package type, but feel free to update it as well for completeness.

To test the build, you can run: `mvn -Dquarkus.package.type=legacy-jar clean package`

**Comment by tthvo** [comment]
> This seems to copy the jar artifact to quarkus-run.jar, no? 

Opps, got mixed up with current one in cryostat-reports that I am looking at that still defines `app.jar`. 

> The base image is indeed out of date. It should use JDK 17 UBI as well. We don't use the legacy-jar package type, but feel free to update it as well for completeness.

Right updated it just now.

**Review by ebaron — COMMENTED** [review]
Hi @tthvo, sorry I thought I submitted this review.

File: src/main/docker/Dockerfile.jvm [hunk 1]
```diff
@@ -18,38 +18,22 @@
...
 
-# Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
-ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
 # We make four distinct layers so if there are application changes the library layers can be re-used
-COPY --chown=1001 target/quarkus-app/lib/ /deployments/lib/
-COPY --chown=1001 target/quarkus-app/*.jar /deployments/
-COPY --chown=1001 target/quarkus-app/app/ /deployments/app/
-COPY --chown=1001 target/quarkus-app/quarkus/ /deployments/quarkus/
+COPY --chown=185 target/quarkus-app/lib/ /deployments/lib/
+COPY --chown=185 target/quarkus-app/*.jar /deployments/
```
[the remark this answers is not in the document]
Comment (tthvo) [reply src/main/docker/Dockerfile.jvm:30]: Ahh thanks I was unsure about this previously too. Let me fix this.

[the remark this answers is not in the document]
Comment (tthvo) [reply src/main/docker/Dockerfile.jvm:30]: Hi sorry, I got a bit of trouble to copy `cryostat-reports-2.3.0-SNAPSHOT.jar` into the image. It seemed not to able to find this jar.

Using:

```docker
COPY --chown=185 target/cryostat-reports-*.jar /deployments/cryostat-reports-runner.jar
```

Error:

```bash
[INFO] [io.quarkus.container.image.docker.deployment.DockerProcessor] Error: error building at STEP "COPY --chown=185 cryostat-reports-2.3.0-SNAPSHOT.jar /deployments/cryostat-reports-runner.jar": checking on sources under "/home/thvo/workspace/cryostat-reports": copier: stat: "/cryostat-reports-2.3.0-SNAPSHOT.jar": no such file or directory
```

[the remark this answers is not in the document]
Comment (tthvo) [reply src/main/docker/Dockerfile.jvm:30]: Just also a question on this. Don't we need to use `target/quarkus-app/quarkus-run.jar` as the argument to `-jar`? I remembered this when upgrading Quarkus versions.

**Review by ebaron — APPROVED** [review]
Looks good! Thanks @tthvo. We should backport this as well.

**Pull request merged**